sorc/ncep_post.fd/INITPOST_NETCDF.f

```diff
@@ -2585,30 +2588,82 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
      if(debugprint)print*,'sample stc = ',1,stc(isa,jsa,9)
 
       END IF
-!
-! E. James - 27 Sep 2022: this is for RRFS, adding smoke and dust
-! extinction; it needs to be after ZINT is defined.
-!
+
       if (modelname == 'FV3R') then
        do l = 1, lm
         do j = jsta_2l, jend_2u
          do i = ista_2l, iend_2u
+          !
+          ! E. James - 27 Sep 2022: this is for RRFS, adding smoke and dust
+          ! extinction; it needs to be after ZINT is defined.
+          !
           if(ext550(i,j,l)<spval)then
             taod5503d ( i, j, l) = ext550 ( i, j, l )
             dz = ZINT( i, j, l ) - ZINT( i, j, l+1 )
             aextc55 ( i, j, l ) = taod5503d ( i, j, l ) / dz
           endif
-         if(debugprint.and.i==im/2.and.j==(jsta+jend)/2)print*,'sample taod5503d= ',   &
+          if(debugprint.and.i==im/2.and.j==(jsta+jend)/2)print*,'sample taod5503d= ',   &
            i,j,l,taod5503d ( i, j, l )
-         if(debugprint.and.i==im/2.and.j==(jsta+jend)/2)print*,'sample dz= ',          &
+          if(debugprint.and.i==im/2.and.j==(jsta+jend)/2)print*,'sample dz= ',          &
            dz
-         if(debugprint.and.i==im/2.and.j==(jsta+jend)/2)print*,'sample AEXTC55= ',     &
+          if(debugprint.and.i==im/2.and.j==(jsta+jend)/2)print*,'sample AEXTC55= ',     &
            i,j,l,aextc55 ( i, j, l )
+
+          ! J. Kenyon - 14 Feb 2024: Obtain the virtual potential
+          ! temperature (thv) from temperature and specific humidity.
+          thv(i,j,l) = ( t(i,j,l) * (p1000/pint(i,j,l))**CAPA ) & ! line 1: convert temp to theta
+                      * ( 1. + 0.61*q(i,j,l)/(1.-q(i,j,l)) )      ! line 2: convert theta to theta-v;
+                                                                  !  note that the factor q/(1-q) converts
+                                                                  !  specific humidity (q) to mixing ratio 
+
          end do
         end do
        end do
+
```

Comment:
@jaymes-kenyon Per UPP code maintenance perspective, I would suggest you created a new subroutine for calculating PBLHGUEST and save as a separate file, example like  CALVIS_GSD.f. Also calculation of PBLHGUST might be called in MDLFLD.f. In that way, this  improvement of 10-m wind gust calculation can be synced in the inline post with the upp submodule upgrade at ufs-weather-model side.

Reply:
Hi @WenMeng-NOAA — thanks for the suggestion!  A few considerations come to mind:

1) If the goal is a "quick fix" prior to the RRFS code freeze (a few weeks away), I think this PR will work.  The PBLHGUST calculation, which has long existed in INITPOST.F, now exists in INITPOST_NETCDF.f.

2) Longer term, I also like the idea of moving the PBLHGUST calculation into a subroutine.  For simplicity, it might be preferable to modify CALPBL.f to provide either a theta-v or a Ri-based calculation, rather than creating another .f file.  But, I might not be able to develop this code before the RRFS deadline.

My suggestion is that we consider this PR as it is.  Then, as a separate issue, I can begin the work of moving the PBLHGUST calculation into a subroutine to improve the overall code.  What do you think?  Thanks again!

Reply:
@jaymes-kenyon Given RRFS code freezing approach, your proposal is working for me.  You might address my comments for RRFS V2.